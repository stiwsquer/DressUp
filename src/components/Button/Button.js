import React from "react";
import { Link, useHistory } from "react-router-dom";
import "./style.scss";

export default function Button({ children, type, nameOfTheClass, onClick }) {
  return (
    <button onClick={onClick} className={nameOfTheClass} type={type}>
      {children}
    </button>
  );
}
