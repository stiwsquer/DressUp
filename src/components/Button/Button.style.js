import styled from "styled-components";

export const StyledButton = styled.button`
  background: #202020;
  color: white;
  border: none;
  outline: none;
  padding: 1rem 3rem;
  letter-spacing: 0.15rem;
  font-weight: 900;
  cursor: pointer;
  &:hover {
    opacity: 0.7;
  }
`;
