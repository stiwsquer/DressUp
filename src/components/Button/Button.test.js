import { BrowserRouter as Router } from "react-router-dom";
import Button from "./Button";

let onButtonClick;
let shallowWrapper;
let mountWrapper;
let renderWrapper;
beforeAll(() => {
  onButtonClick = sinon.spy();
  shallowWrapper = shallow(
    <Button type="submit" className="submitButton" onClick={() => {}} />
  );
  mountWrapper = mount(
    <Button type="submit" className="submitButton" onClick={onButtonClick} />
  );
  renderWrapper = render(
    <Button type="submit" className="submitButton" onClick={() => {}} />
  );
});

it("renders Button component without crashing", () => {
  shallow(<Button />);
});

it("simulates click events", () => {
  mountWrapper.find("button").simulate("click");
  // expect(onButtonClick).toHaveProperty("callCount", 1);
  expect(onButtonClick.calledOnce).toBe(true);
});

it("renders children when passed in", () => {
  const wrapper = shallow(
    <Button>
      <div className="unique" />
    </Button>
  );
  expect(wrapper.contains(<div className="unique" />)).toEqual(true);
});

it("Button snapshot - shallow", () => {
  expect(shallowWrapper).toMatchSnapshot();
});

it("Button snapshot - render", () => {
  expect(renderWrapper).toMatchSnapshot();
});

it("Button snapshot - mount", () => {
  expect(mountWrapper).toMatchSnapshot();
});
