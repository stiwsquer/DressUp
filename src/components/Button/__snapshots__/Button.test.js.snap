// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot - mount 1`] = `
<button
  className="submitButton"
  onClick={[Function]}
  type="submit"
/>
`;

exports[`Button snapshot - render 1`] = `
<button
  class="submitButton"
  type="submit"
/>
`;

exports[`Button snapshot - shallow 1`] = `
<button
  className="submitButton"
  onClick={[Function]}
  type="submit"
/>
`;
